server/middleware/check-client-version.js

```diff
@@ -1,14 +1,22 @@
 import packageJSON from '../../package.json' with { type: "json" };
-const version = packageJSON.version;
 
-export default (req, res, next)=>{
-	const userVersion = req.get('Homebrewery-Version');
+export default (req, res, next) => {
+    const origin = req.get('Origin');
+    const sameSite = req.get('Host');
 
-	if(userVersion != version) {
-		return res.status(412).send({
-			message : `Client version ${userVersion} is out of date. Please save your changes elsewhere and refresh to pick up client version ${version}.`
-		});
-	}
+    if (origin && origin !== `http://${sameSite}` && origin !== `https://${sameSite}`) {
+        return next();  // Skip version check if the request is from another site, like naturalcrit.com
+    }
```

Comment:
I think it would be more appropriate to either move any cross-domain calls before we call `app..use(checkClientVersion)`, or move `checkClientVersion` to be path-specific so it only applies to so database writes where client/server version matters, like "update" or "new" or "delete":

```
router.use('/api', checkClientVersion);
router.post('/api', asyncHandler(api.newBrew));
router.put('/api/:id', asyncHandler(api.getBrew('edit', true)), asyncHandler(api.updateBrew));
router.put('/api/update/:id', asyncHandler(api.getBrew('edit', true)), asyncHandler(api.updateBrew));
router.delete('/api/:id', asyncHandler(api.deleteBrew));
router.get('/api/remove/:id', asyncHandler(api.deleteBrew));
router.get('/api/theme/:renderer/:id', asyncHandler(api.getThemeBundle));
```

becomes

```
router.post('/api', checkClientVersion, asyncHandler(api.newBrew));
router.put('/api/:id', checkClientVersion, asyncHandler(api.getBrew('edit', true)), asyncHandler(api.updateBrew));
router.put('/api/update/:id', checkClientVersion, asyncHandler(api.getBrew('edit', true)), asyncHandler(api.updateBrew));
router.delete('/api/:id', checkClientVersion, asyncHandler(api.deleteBrew));
router.get('/api/remove/:id', checkClientVersion, asyncHandler(api.deleteBrew));
router.get('/api/theme/:renderer/:id', asyncHandler(api.getThemeBundle));
```

Reply:
This makes sense too, got it